src/ui/components/trinary_input_string.c

```diff
@@ -235,6 +235,31 @@ static void _render(component_t* component)
     }
 }
 
+// if the current input uniquely identifies a word from the wordlist by prefix, we autocomplete the
+// word.
+static void _maybe_autocomplete(component_t* trinary_input_string)
+{
+    data_t* data = (data_t*)trinary_input_string->data;
+    if (data->wordlist == NULL) {
+        return;
+    }
+    // initial value means no word was found yet.
+    size_t found_word_idx = data->wordlist_size;
+    for (size_t word_idx = 0; word_idx < data->wordlist_size; word_idx++) {
+        const char* word = data->wordlist[word_idx];
+        bool is_prefix = strncmp(data->string, word, data->string_index) == 0;
+        if (is_prefix) {
+            if (found_word_idx != data->wordlist_size) {
+                // Not unique.
+                return;
+            }
+            found_word_idx = word_idx;
+        }
+    }
+    data->string_index =
```

Comment:
could think about getting rid of string_index completely, and use `strlen(data->string)` in its place everywhere.